samcli/lib/providers/provider.py

```diff
@@ -38,7 +38,7 @@ class Function(NamedTuple):
     codeuri: Optional[str]
     # Environment variables. This is a dictionary with one key called Variables inside it.
     # This contains the definition of environment variables
-    environment: Optional[str]
+    environment: Optional[Dict]
```

Comment:
Would this create any issue with backward compatibility? If it was `Dict` when we first introduced types, why it didn't fail at that time but we are updating it with `Dict` now?

Reply:
type annotations have no runtime impact and are only checked in CI. 
This was fixing my original type annotation. Should have been `Dict` in the first place 😓